include/seqan3/alphabet/aminoacid/translation.hpp

```diff
@@ -0,0 +1,187 @@
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup aminoacid
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Contains functions for translating a triplet of nucleotides into an amino acid
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/range/concept.hpp>
+
+namespace seqan3
+{
+/*!\brief Genetic codes used for translation of nucleotides into amino acids
+ *
+ * \details
+ * The numeric values of the enums correspond to the genbank transl_table values
+ * (see "http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi").
+ */
+enum struct genetic_code : uint8_t
+{
+    CANONICAL=1,
+//     VERT_MITOCHONDRIAL,
+//     YEAST_MITOCHONDRIAL,
+//     MOLD_MITOCHONDRIAL,
+//     INVERT_MITOCHONDRIAL,
+//     CILIATE,
+//     FLATWORM_MITOCHONDRIAL = 9,
+//     EUPLOTID,
+//     PROKARYOTE,
+//     ALT_YEAST,
+//     ASCIDIAN_MITOCHONDRIAL,
+//     ALT_FLATWORM_MITOCHONDRIAL,
+//     BLEPHARISMA,
+//     CHLOROPHYCEAN_MITOCHONDRIAL,
+//     TREMATODE_MITOCHONDRIAL = 21,
+//     SCENEDESMUS_MITOCHONDRIAL,
+//     THRAUSTOCHYTRIUM_MITOCHONDRIAL,
+//     PTEROBRANCHIA_MITOCHONDRIAL,
+//     GRACILIBACTERIA
+};
+}
+
+#include <seqan3/alphabet/aminoacid/translation_details.hpp>
+
+namespace seqan3
+{
+/*!\brief Translate one nucleotide triplet into single amino acid (single nucleotide interface)
+ *
+ * \param n1 First nucleotide in triplet
+ * \param n2 Second nucleotide in triplet
+ * \param n3 Third nucleotide in triplet
+ *
+ * \details
+ *
+ * Translates single nucleotides into amino acid according to given genetic code.
+ *
+ * \par Complexity
+ *
+ * Constant.
+ *
+ * \par Exceptions
+ *
+ * No-throw guarantee.
+*/
+template <nucleotide_concept nucl_type, genetic_code gc = genetic_code::CANONICAL>
```

Comment:
Can you put the genetic_code as first template argument? The reason is that you want to be able to do:
```cpp
aa27 a = translate_triplet<genetic_code::FLATWORM>(n1, n2, n3);
```
If you put the nucl_type first, you force the user to explicitly state the type (template arguments have to be specified from left to right, even if some can be deduced).